File: app/src/main/java/com/wojdor/hearthcards/application/util/HtmlParser.java
package com.wojdor.hearthcards.application.util;

import android.text.Html;
import android.text.Spanned;

public class HtmlParser {

    private static final String EMPTY = "";
    private static final String ENTER = "\n";
    private static final String ENTER_SIGN = "\\n";
    private static final String ENTER_HTML = "<br>";
    private static final String DOLLAR_SIGN = "$";

    public Spanned asHtml(String text) {
        if (text == null) return Html.fromHtml(EMPTY);
        String editedText = text.replace(ENTER_SIGN, ENTER_HTML).replace(DOLLAR_SIGN, EMPTY);
        return Html.fromHtml(editedText);
    }

    public String asString(String text) {
        if (text == null) return EMPTY;
        return Html.fromHtml(text).toString().replace(ENTER_SIGN, ENTER).replace(DOLLAR_SIGN, EMPTY);
    }
}

File: app/src/main/java/com/wojdor/hearthstonecards/application/util/HtmlParser.kt
package com.wojdor.hearthstonecards.application.util

import android.text.Html
import android.text.Spanned

class HtmlParser {

    fun asHtml(text: String?): Spanned {
        if (text == null) return Html.fromHtml(EMPTY)
        val editedText = text
                .replace(ENTER_SIGN, ENTER_HTML)
                .replace(DOLLAR_SIGN, EMPTY)
                .replace(UNDERLINE, SPACE)
                .replace(WEIRD_X_SYMBOL, EMPTY)
        return Html.fromHtml(editedText)
    }

    fun asString(text: String?): String {
        return if (text == null) EMPTY else Html.fromHtml(text).toString()
                .replace(ENTER_SIGN, ENTER)
                .replace(DOLLAR_SIGN, EMPTY)
                .replace(UNDERLINE, SPACE)
                .replace(WEIRD_X_SYMBOL, EMPTY)
    }

    companion object {
        private const val EMPTY = ""
        private const val SPACE = " "
        private const val ENTER = "\n"
        private const val ENTER_SIGN = "\\n"
        private const val ENTER_HTML = "<br>"
        private const val DOLLAR_SIGN = "$"
        private const val UNDERLINE = "_"
        private const val WEIRD_X_SYMBOL = "[x]"
    }
}
